mmv1/products/dialogflow/Generator.yaml

```
@@ -0,0 +1,248 @@
+# Copyright 2025 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'Generator'
+description: |
+  LLM generator.
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/agent-assist/docs'
+  api: 'https://cloud.google.com/dialogflow/es/docs/reference/rest/v2beta1/projects.locations.generators'
+base_url: 'projects/{{project}}/locations/{{location}}/generators'
+create_url: 'projects/{{project}}/locations/{{location}}/generators?generatorId={{generator_id}}'
+self_link: '{{name}}'
+update_verb: 'PATCH'
+update_mask: true
+
+custom_code:
+  pre_create: 'templates/terraform/pre_create/dialogflow_set_endpoint.go.tmpl'
+  pre_read: 'templates/terraform/pre_create/dialogflow_set_endpoint.go.tmpl'
+  pre_update: 'templates/terraform/pre_create/dialogflow_set_endpoint.go.tmpl'
+  pre_delete: 'templates/terraform/pre_create/dialogflow_set_endpoint.go.tmpl'
+  post_create: 'templates/terraform/post_create/set_computed_name.tmpl'
+  custom_import: 'templates/terraform/custom_import/self_link_as_name_set_project_and_location.go.tmpl'
+
+examples:
+  - name: 'dialogflow_generator_basic'
+    primary_resource_id: 'summarization_basic_generator'
+    vars:
+      profile_name: 'dialogflow-profile'
+
+parameters:
+  - name: 'location'
+    type: String
+    required: true
+    url_param_only: true
+    description: |
+      desc
+  - name: 'generatorId'
+    type: String
+    default_from_api: true
+    custom_flatten: 'templates/terraform/custom_flatten/id_from_name.tmpl'
+    description: |
+      Optional. The ID to use for the generator, which will become the final component of the generator's resource name.
+
+properties:
+  - name: name
+    type: String
+    description: |
+      The resource name of the generator.
+    output: true
+  - name: description
+    type: String
+    description: |
+      Optional. Human readable description of the generator.
+  - name: summarizationContext
+    type: NestedObject
```

```suggestion
    type: NestedObject
    required: true
```

Based on offline chat, this is part of a required union field. Marking this as required for now. We can change it to [exactly_one_of](https://googlecloudplatform.github.io/magic-modules/reference/field/#exactly_one_of) once other  fields are implemented.